File: Array/sort-an-array-of-0s-1s-and-2s.py
# https://practice.geeksforgeeks.org/problems/sort-an-array-of-0s-1s-and-2s4231/1


def sort012(a,n):
	zeroc=0
	onec=0
	twoc=0
	res=[]
	for i in range(n):
		if a[i]==0:
			zeroc+=1
		elif a[i]==1:
			onec+=1
		elif a[i]==2:
			twoc+=1
	while zeroc:
		res.append(0)
		zeroc-=1
	while onec:
		res.append(1)
		onec-=1
	while twoc:
		res.append(2)
		twoc-=1
	print(res)
		


if __name__=='__main__':
	a = [0, 2, 1, 2, 0]
	n=len(a)
	sort012(a,n)
	
	a.sort()
	print(a)
	
	a.sort(reverse=True)
	print(a)

File: Array/sort-an-array-of-0s-1s-and-2s.cpp
//https://practice.geeksforgeeks.org/problems/sort-an-array-of-0s-1s-and-2s/0
// can also use any sorting algorithm

using namespace std;
int main() {

int t;
cin>>t;
while(t--){
		   long long int n;
		   cin>>n;
		   
		   int count0=0;
		   int count1=0;
		   int count2=0;
		   int num;
		   for(long long int i=0;i<n;i++ ){
		   	cin>>num;
		   	if(num==0)
		   			  count0++;
		   	
		   	if(num==1)
		   			  count1++;
			
		   	if(num==2)
		   			  count2++;			 		  
		   			  
		   }
		   
		   while(count0--){
		   	
		   	cout<<"0"<<" ";
		   }
		   
		   while(count1--){
		   	
		   	cout<<"1"<<" ";
		   }
		   
		   while(count2--){
		   	
		   	cout<<"2"<<" ";
		   }

	
	
	cout<<endl;
}



	return 0;
}